app/resources/views.py

```diff
@@ -45,18 +46,76 @@ def add():
                             user_id=int(current_user.get_id()))
         db.session.add(resource)
         db.session.commit()
-        return redirect(url_for('resources.index'))
+        return redirect(url_for('resources.show', resource_id=resource.id))
     return render_template('resources/add_resource.html', form=form)
 
 
 @resources.route('/resource/<int:resource_id>')
 @login_required
-def show_resource(resource_id):
-    """
-    Show the resource with the given id, the id is an integer.
-    """
+def show(resource_id):
     resource = Resource.query.get_or_404(resource_id)
     address = Address.query.get(resource.address_id)
     user = User.query.get(resource.user_id)
+    current_user_id = int(current_user.get_id())
+    # TODO: base template for reviews.
     return render_template('resources/view_resource.html', resource=resource,
-                           address=address, user=user)
+                           address=address, user=user,
+                           current_user_id=current_user_id)
+
+
+@resources.route('/resource/<int:resource_id>/writeareview',
+                 methods=['GET', 'POST'])
+@login_required
+def review(resource_id):
+    resource = Resource.query.get_or_404(resource_id)
+    address = Address.query.get(resource.address_id)
+    user = User.query.get(resource.user_id)
+    form = ReviewForm()
+    if form.validate_on_submit():
+        review = ResourceReview(timestamp=datetime.now(),
+                                content=form.content.data,
+                                rating=form.rating.data,
+                                resource_id=resource_id,
+                                user_id=int(current_user.get_id()))
+        db.session.add(review)
+        db.session.commit()
+        return redirect(url_for('resources.show', resource_id=resource.id))
+    return render_template('resources/write_review.html', resource=resource,
+                           address=address, user=user, form=form)
+
+
+@resources.route('/resource/<int:resource_id>/deleteareview/<int:review_id>')
+@login_required
+def delete(resource_id, review_id):
+    # TODO: test that random user can't delete another user's review.
```

Comment:
should be pretty easy to do here no (after you query for the review)? something like this?
```python
if (current_user.id != review.user.id):
    tell_user_no()
```